pkg/controller/cachemanager/cachemanager.go

```diff
@@ -0,0 +1,417 @@
+package cachemanager
```

Comment:
Why have we moved to `pkg/controller`?

Reply:
since the `cachemanager` is not a controller, it doesn't make 100% sense to have it here but I think initially we decided to have this structure to better present the separation of responsibility.

we're gonna wait/ see what other maintainers think.

Reply:
Best to only have controllers (or meta controllers or abstract controllers) in controllers

Reply:
Sometimes we put sub-libraries of a controller in here (`.../controller/config/process`, for example, but I don't think this is owned by a specific controller.

Reply:
undid this